snapshots/overlay/overlay.go

```diff
@@ -237,11 +237,15 @@ func (o *snapshotter) Mounts(ctx context.Context, key string) ([]mount.Mount, er
 		return nil, err
 	}
 	s, err := storage.GetSnapshot(ctx, key)
+	_, info, _, errInfo := storage.GetInfo(ctx, key)
 	t.Rollback()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get active mount")
 	}
-	return o.mounts(s), nil
+	if errInfo != nil {
+		return nil, errors.Wrap(errInfo, "failed to get snapshot info")
+	}
```

Comment:
Is there any specific reason why this check is not just after `_, info, _, errInfo := storage.GetInfo(ctx, key)`?

Reply:
Yes, thanks for comment. It's better to check errors exactly after the function call and just introduce deferred t.Rollback() call. Otherwise in case of error in `s, err := storage.GetSnapshot(ctx, key)` the `_, info, _, err := storage.GetInfo(ctx, key)` will be executed unconditionally. Let's do like this:
```
func (o *snapshotter) Mounts(ctx context.Context, key string) ([]mount.Mount, error) {
        ctx, t, err := o.ms.TransactionContext(ctx, false)
        if err != nil {
                return nil, err
        }

        defer func() {
                if err != nil {
                        t.Rollback()
                }
        }()
        s, err := storage.GetSnapshot(ctx, key)
        if err != nil {
                return nil, errors.Wrap(err, "failed to get active mount")
        }
        _, info, _, err := storage.GetInfo(ctx, key)
        if err != nil {
                return nil, errors.Wrap(err, "failed to get snapshot info")
        }
        t.Rollback()

        return o.mounts(s, info), nil
}